docs/reference/end_to_end_planning.md

```diff
@@ -0,0 +1,157 @@
+# end_to_end_planning module
+
+The *end_to_end_planning* module contains the *EndToEndPlanningRLLearner* class, which inherits from the abstract 
+class *LearnerRL*.
+
+### Class EndToEndPlanningRLLearner
+Bases: `engine.learners.LearnerRL`
+
+The *EndToEndPlanningRLLearner* is an agent that can be used to train quadrotor robots equipped with a depth sensor to
+follow a provided trajectory while avoiding obstacles.
+
+The [EndToEndPlanningRLLearner](/src/opendr/planning/end_to_end_planning/e2e_planning_learner.py) class has the 
+following public methods:
+
+#### `EndToEndPlanningRLLearner` constructor
+
+Constructor parameters:
+
+- **env**: *gym.Env*\
+  Reinforcment learning environment to train or evaluate the agent on.
+- **lr**: *float, default=3e-4*\
+  Specifies the initial learning rate to be used during training.
+- **n_steps**: *int, default=1024*\
+  Specifies the number of steps to run for environment per update.
+- **iters**: *int, default=5e4*\
+  Specifies the number of steps the training should run for.
+- **batch_size**: *int, default=64*\
+  Specifies the batch size during training.
+- **checkpoint_after_iter**: *int, default=500*\
+  Specifies per how many training steps a checkpoint should be saved.
+- **temp_path**: *str, default=''*\
+  Specifies a path where the algorithm stores log files and saves checkpoints.
+- **device**: *{'cpu', 'cuda'}, default='cuda'*\
+  Specifies the device to be used.
+
+#### `EndToEndPlanningRLLearner.fit`
+```python
+EndToEndPlanningRLLearner.fit(self, env, logging_path, silent, verbose)
+```
+
+Train the agent on the environment.
+
+Parameters:
+
+- **env**: *gym.Env, default=None*\
+  If specified use this env to train.
+- **logging_path**: *str, default=''*\
+  Path for logging and checkpointing.
+- **silent**: *bool, default=False*\
+  Disable verbosity.
+- **verbose**: *bool, default=True*\
+  Enable verbosity.
+
+
+#### `EndToEndPlanningRLLearner.eval`
+```python
+EndToEndPlanningRLLearner.eval(self, env)
+```
+Evaluate the agent on the specified environment.
+
+Parameters:
+
+- **env**: *gym.Env, default=None*\
+  Environment to evaluate on.
+
+
+#### `EndToEndPlanningRLLearner.save`
+```python
+EndToEndPlanningRLLearner.save(self, path)
+```
+Saves the model in the path provided.
+
+Parameters:
+
+- **path**: *str*\
+  Path to save the model, including the filename.
+
+
+#### `EndToEndPlanningRLLearner.load`
+```python
+EndToEndPlanningRLLearner.load(self, path)
+```
+Loads a model from the path provided.
+
+Parameters:
+
+- **path**: *str*\
+  Path of the model to be loaded.
+
+
+
+
+### Simulation environment setup
+
+The environment includes an Ardupilot controlled quadrotor in Webots simulation. 
+Installation of Ardupilot software follows this link: `https://github.com/ArduPilot/ardupilot`
+
+The files under `ardupilot.zip` should be replaced under the installation of Ardupilot, can be downloaded by running 
+`download_ardupilot_files.py` script.
```

Comment:
This phrase sounds a bit odd. The utility downloads the zip and the user should replace the contents?

Reply:
Rephrased.